intelligencelayer/shared/superresolution.py

```diff
@@ -0,0 +1,184 @@
+import argparse
+import cv2
+import numpy as np
+import os
+import sys
+import torch
+import json
+from multiprocessing import Process
+import base64
+import time
+
+
+from superresolution.utils import load_model
+from superresolution.common import tensor2img, calculate_psnr, calculate_ssim, bgr2ycbcr
+
+
+sys.path.insert(0, os.path.join(os.path.dirname(os.path.realpath(__file__)), "."))
+
+
+from shared import SharedOptions
+if SharedOptions.PROFILE == "windows_native":
+    sys.path.append(os.path.join(SharedOptions.APP_DIR,"windows_packages"))
+
+
+def get_network(model_path):
+    from superresolution.config import config
+    from superresolution.network import Network
+
+    # an ugly operation
+    if 'KERNEL_PATH' in config.MODEL:
+        config.MODEL.KERNEL_PATH = config.MODEL.KERNEL_PATH.replace('../', '')
+
+    return config, Network(config)
+
+def superresolution4x(thread_name: str, delay: float):
+    SHARED_APP_DIR = SharedOptions.SHARED_APP_DIR
+    CUDA_MODE = SharedOptions.CUDA_MODE
+    db = SharedOptions.db
+    TEMP_PATH = SharedOptions.TEMP_PATH
+
+    IMAGE_QUEUE = "superresolution_queue"
+
+    model_path = os.path.join(
+            SHARED_APP_DIR, SharedOptions.SETTINGS.SUPERRESOLUTION_MODEL
+        )
+
+    use_cuda = SharedOptions.CUDA_MODE
+    use_cpu = False if use_cuda else True
+
+    print('Loading Network ...')
+    config, model = get_network(model_path)
+    if use_cuda:
+        device = torch.device('cuda')
+    else:
+        device = torch.device('cpu')
+
+    model = model.to(device)
+    load_model(model, model_path, strict=True, cpu=use_cpu)
+
+    down = config.MODEL.DOWN
+    scale = config.MODEL.SCALE
+
+
+    while True:
+        queue = db.lrange(IMAGE_QUEUE, 0, 0)
+
+        db.ltrim(IMAGE_QUEUE, len(queue), -1)
+
+        if len(queue) > 0:
+
+            for req_data in queue:
+                try:
+                    with torch.no_grad():
+                        req_data = json.JSONDecoder().decode(req_data)
+                        img_id = req_data["imgid"]
+                        req_id = req_data["reqid"]
+                        req_type = req_data["reqtype"]
+                        img_path = os.path.join(TEMP_PATH, img_id)
+
+                        lr_img = cv2.imread(img_path, cv2.IMREAD_COLOR)

```

Comment:
cv2 needs to be replaced by PIL here. This can be written as below
``` 
lr_img = Image.open(img_path).convert("RGB") 
lr_img = np.asarray(lr_img)
```